ast: don't add semicolons to static assertion blocks

theofabilous opened the pull request:
Closes #220

`_Static_assert`s inside function bodies are rendered with a trailing semicolon, yielding syntactically invalid zig code.

Fixed by not appending a semicolon to static assertions in `addSemicolonIfNeeded()`

Vexu commented:
Looks like I accidentally dropped that commit on rebase. I'll use this if you rebase it after  #252

theofabilous commented:
> Looks like I accidentally dropped that commit on rebase. I'll use this if you rebase it after #252

Sounds good.

theofabilous commented:
Rebased onto main now that #252 is merged. I also took the liberty of applying and squashing the `_Static_assert` test modifications in https://github.com/ziglang/translate-c/commit/836ea03f4acbe535a4e859f547354b05ab22d766 and added you as a co-author
